### PR TITLE
Data processor train() and eval() more explicit

### PR DESCRIPTION
**Pull request by dhpitt**
`DefaultDataProcessor` currently has behavior involving `self.train` (which is currently a bound method of `DataProcessor` inherited from `nn.Module`). This fixes that and also sets data procesor train and eval by default within the `Trainer`.

### CODE REVIEW

File: neuralop/datasets/data_transforms.py [hunk 1]
```diff
@@ -108,6 +114,12 @@ def forward(self, **data_dict):
         output = self.model(data_dict["x"])
         output = self.postprocess(output)
         return output, data_dict
+    
+    def train(self):
+        self._train = True
+    
+    def train(self):
+        self._train = False
```
Comment (JeanKossaifi) [inline neuralop/datasets/data_transforms.py:122]: I don't get these two functions - you're redefining? 
We should define `train` as a property. But if it's an nn.Module why not reuse the existing one?

File: neuralop/datasets/data_transforms.py [hunk 2]
```diff
@@ -45,6 +45,11 @@ def wrap(self, model):
     def forward(self, x):
         pass
 
+    def train(self):
```
Comment (JeanKossaifi) [inline neuralop/datasets/data_transforms.py:48]: I don't see the need for this class. Not only does it not bring feature it actually breaks the default behaviour of the functions (it should at the very least call super). I'd be in favor of removing it alltogether.

Comment (dhpitt) [reply neuralop/datasets/data_transforms.py]: update

File: neuralop/datasets/data_transforms.py (hunk 1 above)
Comment (dhpitt) [reply neuralop/datasets/data_transforms.py]: Good idea

File: neuralop/data/transforms/data_processors.py [hunk 3]
```diff
@@ -108,6 +112,17 @@ def forward(self, **data_dict):
         output = self.model(data_dict["x"])
         output = self.postprocess(output)
         return output, data_dict
+        
+    def train(self, val: bool=True):
+        super().train(val)
+        if self.model:
```
Comment (JeanKossaifi) [inline neuralop/data/transforms/data_processors.py:118]: should this be "is not None"?

**Review by JeanKossaifi — COMMENTED** [review]
Looks good, just one minor point.

File: neuralop/data/transforms/data_processors.py (hunk 3 above)
Comment (dhpitt) [reply neuralop/data/transforms/data_processors.py]: Changed.

**Comment by JeanKossaifi** [comment]
Looks good, merging! Thanks @dhpitt !

**Pull request merged**